src/test/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionServiceTests.java

```diff
@@ -197,8 +206,33 @@ public void shouldGetAppPalaemonTokens() throws Exception {
                 .isEqualTo(Map.of(
                     IexecEnvUtils.IEXEC_INPUT_FILE_NAME_PREFIX + "1", "file1",
                     IexecEnvUtils.IEXEC_INPUT_FILE_NAME_PREFIX + "2", "file2"));
+        assertThat(tokens).containsEntry(APP_PROVIDER_SECRET + "_0", SECRET_VALUE);
```

Comment:
How about defining `"APP_PROVIDER_SECRET_0"` inside the test itself instead of using `ApplicationRuntimeSecretService#APP_PROVIDER_SECRET`. This makes sure we don't break the contract with applications without noticing it.